bot.py

```diff
@@ -0,0 +1,192 @@
+from teams import Team
+import chess_utils
+import pieces
+
+
+def try_castling(white_team, bot_team):
+    king = None
+    rook1_square = None
+    rook2_square = None
+    for piece in bot_team.pieces:
+        if isinstance(piece, pieces.King):
+            king = piece
+            continue
+
+        if isinstance(piece, pieces.Rook):
+            if rook1_square is None:
+                rook1_square = piece.square
+                continue
+            rook2_square = piece.square
+
+    if rook1_square is not None:
+        try:
+            chess_utils.try_to_move(king, rook1_square, bot_team, white_team)
+            return True, king
+        except chess_utils.MoveError:
+            pass
+
+    if rook2_square is not None:
+        try:
+            chess_utils.try_to_move(king, rook2_square, bot_team, white_team)
+            return True, king
+
+        except chess_utils.MoveError:
+            pass
+
+    return False, king
+
+
+def move(white_team: Team, bot_team: Team, depth=2):
+    is_castling, king = try_castling(white_team, bot_team)
+    if is_castling:
+        return king
+
+    if chess_utils.is_checkmated(bot_team, white_team) or chess_utils.is_tie(bot_team, white_team):
+        return None
+
+    score, best_move = mini(white_team, bot_team, depth)
+
+    print(f'piece - {best_move[0]}. moved to - {best_move[1]}\n'
+          f'current score after the move is {score}')
+
+    piece, move_square = best_move
+    chess_utils.try_to_move(piece, move_square, bot_team, white_team)
+    return piece
+
+
+def mini(white_team: Team, bot_team: Team, depth):
+    best_score = 1000
+    best_move = None
+
+    if chess_utils.is_tie(bot_team, white_team):
+        return 0
+
+    if depth == 0:
+        return chess_utils.get_score(white_team, bot_team), best_move
+
+    for piece in bot_team.pieces:
+        if piece.is_eaten:
+            continue
+        valid_moves = piece.get_valid_move_squares()
+        for move_square in valid_moves:
+            try:
+                # If Didn't move, code wouldn't crash, just move to next move.
+                score_after_move = futuire_move(piece, move_square, white_team, bot_team, depth, is_bot_futiure_turn=True)
```

Comment:
Grammar issue - fix futuire to future everywhere.

Reply:
:+1: